### ISSUE TITLE
Enchantments not working properly

### ISSUE THREAD

**Issue by Uboaaaaa** [Geolykt/EnchantmentsPlus#16]
Tested on git-paper-231 1.16.3-R0.1-SNAPSHOT

- There's no Apocalyse, whilst Anthropomorphism unleashes hell instead.
- Lumber seems not working.
- Bow and sword has Stationary, entities still got pushed back.
- Switch switches two blocks at once
- Weight alters player walking speed whilst giving them slowness effects. The walking speed adjustment is not restored after unequipping the gear.

Some others issues that might not be as significant and important:
- Mow doesn't consume durability
- When used on ground of the same level, Terraformer easily stuck the player in the blocks they just formed.

**Comment by Geolykt** [comment]
What's the version of the plugin?
However it does seem that your changes seem to make sense in current versions, so they'll be fixed rather soon (hopefully).

**Comment by Uboaaaaa** [comment]
Sorry forgot to include the plugin version: 2.1.5c

**Comment by Uboaaaaa** [comment]
Yes these issues still persists in 2.1.5d.

**Comment by Geolykt** [comment]
2.1.5d doesn't change anything regarding this topic; the v2.1.5 (although pretty much every v2.1.x) update has been suprisingly unstable, but hopefully I'll get it under controll rather quick.

I'm actually suprised by the amount of people that had the original plugin, I expected to see no more than 3 people, but now I have a community of around a dozen of people attracted. It's funny to think back that the fork started just as a simple Bugfix for the fire enchantment, only to continue as something much greater

**Comment by Geolykt** [comment]
Initial reproduction report:
 - Anthropomorphism is bugged indeed and this issue has been fixed, a workaround is to change the getter to a NBT getter (does more harm than good though)
 - The lumber issue is known, why exactly this occours needs to be inspected further but it persists since at least v1.2.0 (9577d2e45b93c19d1d7d918aa8ad8087b6432825). I guess that the default configurations are missing something or there's another issue
 - Stationary is confirmed to work (however has some issues with tipped arrows - which we'd need to fix)
 - The issue with switch was also reproduceable and also exists since at least v1.2.0 (to be for to myself, I dismissed them as probably working as intended, with hindsight that wasn't totally right), but they would also be an easy fix
 - The weight issue should have been fixed already, but I forgot to remove the responsible old code, which also has been fix.
 - I'm not entirely sure on what to say about the durabillity exploits (they exist), since they also exist in the old plugin should they be fixed or nah?
 - The Terraformer issue has also been existing forever and would also be a rather easy fix

So in short what needs to be and has been fixed:

- [ x ] Anthromorphism acting like Apocalypse
- [ ] Lumber enchantment now working all the time as intended
- [ ] Class cast exception when using tipped arrows
- [ ] Switch switching double the time
- [ x ] Weight issue
- Durabillity exploits (too many to list)
- [ ] Terraformer enchantment can trap the own player

**Comment by Geolykt** [comment]
The Lumber enchantment actually appears to work. Did you forget to shift?

**Comment by Uboaaaaa** [comment]
Definitely forgot that. Sorry that was my bad.

**Comment by Geolykt** [comment]
Aside from the Durabillity exploits every issue has now been solved and will be fixed within v2.1.6 which should be released later today
